gtd_generative_proof_machine/src/generative_proof_machine.c

```diff
@@ -0,0 +1,152 @@
+#include "generative_proof_machine.h"
+#include "physical_graph.h"
+#include "generative_restriction.h"
+#include "common.h"
+
+typedef struct GenerativeProofMachine
+{
+    Graph *graph;
+    GenerativeRestriction **restrictions;
+    int numRestrictions;
+    int depth;
+    ProofTree *proofTree;
+} GenerativeProofMachine;
+
+/**
+ * \brief constructor for GenerativeProofMachine class
+ * \param numRestrictions number of generative restrictions
+ * \param startGraph start graph
+ * \returns pointer to a newly created GenerativeProofMachine
+ */
+GenerativeProofMachine *create_proof_machine(int numRestrictions, Graph *startGraph)
+{
+    GenerativeProofMachine *machine = gtd_malloc(sizeof(GenerativeProofMachine));
```

Comment:
good practice is to always cast type returned from malloc (does not have to be changed everywhere now, do not waste time)